packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ISharedObject } from "@fluidframework/shared-object-base/internal";
+import {
+	type ImplicitFieldSchema,
+	type InsertableTreeFieldFromImplicitField,
+	type ITree,
+	SharedTree,
+	type TreeView,
+	type TreeViewConfiguration,
+} from "@fluidframework/tree/internal";
+
+import { PureDataObject } from "./pureDataObject.js";
+
+/**
+ * Channel ID of {@link TreeDataObject}'s root {@link @fluidframework/tree#SharedTree}.
+ */
+const treeChannelId = "root";
+
+/**
+ * {@link @fluidframework/tree#SharedTree}-backed {@link PureDataObject | data object}.
+ *
+ * @internal
+ */
+export abstract class TreeDataObject<
```

Comment:
I see a decent amount of code changes including new classes like this one in this package, but no test changes. I think we should include some unit tests for the new code, or at the very least a TODO to do so so people looking at this code later (ex thinking about exposing it more publicially) will see it is in need of tests

Reply:
No, that's a good call. I'll add some test coverage before merging.

Reply:
Alright, this was much less trivial than I was expecting. `aqueduct` currently has no tests, and that appears to be due to a lack of reasonable infrastructure for doing so. Well, at least in places that `aqueduct` can use.

For now, I added a couple of tests in `test-end-to-end-tests`, with a note that they really belong in the `aqueduct` package.